File: src/ProjectR/content/scripts/spells/MagicCharge.lua
function GetTargetType()
    return Single
end

function GetName()
    return "Magic Charge"
end

function GetDescription()
    return "Surround yourself in a magic shield and charge at your target. Deals damage according to yoour MR."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 0
end

function GetDelay()
  return .5
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
    local damage = aMR - dMR
    d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/MagicCharge.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class MagicCharge : SpellScriptBase
    {
        public override string Name { get { return "Magic Charge"; } }
        public override string Description { get { return "Surround yourself in a magic shield and charge at your target. Deals damage according to your MR."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 0; } }
        public override double Delay { get { return .5; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            DealDamage(cMR - tMR);
        }
    }
}